src/app/GitUI/LeftPanel/FavoritesTree.cs

```diff
@@ -0,0 +1,168 @@
+            }
+        }
+
+        private void CreateLocalBranchTree(IGitRef branch, Dictionary<string, BaseRevisionNode> pathToNodes, Nodes nodes, IDictionary<string, AheadBehindData>? aheadBehindData)
+        {
+            Validates.NotNull(branch.ObjectId);
+
+            if (!_cachedFavorites.Contains(branch.ObjectId.ToString()))
+            {
+                return;
+            }
+
+            string currentBranch = _revisionGridInfo.GetCurrentBranch();
+
+            LocalBranchNode branchNode = new(this, branch.ObjectId, branch.Name, branch.Name == currentBranch, true);
+
+            if (aheadBehindData?.TryGetValue(branchNode.FullPath, out AheadBehindData aheadBehind) is true)
+            {
+                branchNode.UpdateAheadBehind(aheadBehind.ToDisplay(), aheadBehind.RemoteRef);
+            }
+
+            BaseRevisionNode parent = branchNode.CreateRootNode(pathToNodes, CreatePathNode);
+
+            if (parent is not null)
+            {
+                nodes.AddNode(parent);
+            }
+
+            BaseRevisionNode CreatePathNode(Tree tree, string parentPath)
+            {
+                BranchPathNode branchPathNode = new(tree, parentPath);
+
+                return branchPathNode;
+            }
+        }
+
+        private void CreateRemoteBranchTree(IGitRef branch, Dictionary<string, BaseRevisionNode> pathToNodes, Nodes nodes, IDictionary<string, AheadBehindData>? aheadBehindData)
+        {
+            Validates.NotNull(branch.ObjectId);
+
+            if (!_cachedFavorites.Contains(branch.ObjectId.ToString()))
+            {
+                return;
+            }
+
+            ConfigFileRemoteSettingsManager remotesManager = new(() => Module);
+            Dictionary<string, Remote> remoteByName = ThreadHelper.JoinableTaskFactory.Run(Module.GetRemotesAsync).ToDictionary(r => r.Name);
+
+            string remoteName = branch.Name.SubstringUntil('/');
+
+            if (remoteByName.TryGetValue(remoteName, out Remote remote))
+            {
+                RemoteBranchNode branchNode = new(this, branch.ObjectId, branch.Name, true);
+
+                if (aheadBehindData?.TryGetValue(branch.CompleteName, out AheadBehindData aheadBehind) is true)
+                {
+                    branchNode.UpdateAheadBehind(aheadBehind.ToDisplay(), $"{GitRefName.RefsHeadsPrefix}{aheadBehind.Branch}");
+                }
+
+                BaseRevisionNode parent = branchNode.CreateRootNode(pathToNodes, (tree, parentPath) => CreateRemoteBranchPathNode(tree, parentPath, remote));
+
+                if (parent is not null)
+                {
+                    nodes.AddNode(parent);
+                }
+            }
+
+            BaseRevisionNode CreateRemoteBranchPathNode(Tree tree, string parentPath, Remote remote)
+            {
+                if (parentPath == remote.Name)
+                {
+                    return new RemoteRepoNode(tree, parentPath, remotesManager, remote, true);
```

Comment:
Please use named arguments everywhere where literals are passed.
```suggestion
                    return new RemoteRepoNode(tree, parentPath, remotesManager, remote, isEnabled: true);
```